src/htr2hpc/api_client.py

```diff
@@ -0,0 +1,329 @@
+        """Update an existing model record with a new model file."""
+        api_url = f"models/{model_id}/"
+        with open(model_file, "rb") as mfile:
+            files = {"file": mfile}
+            data = {
+                "name": "updated model",
+                "job": "Segment",  # required; get from existing record?
+                # file_size (int)  - set from pathlib object using .stat().st_size
+                # versions ?
+                # accuracy_percent
+            }
+            resp = self._make_request(api_url, method="PUT", files=files, data=data)
+        # on successful update, returns the model object
+        return to_namedtuple("model", resp.json())
+
+    def documents(self):
+        """paginated list of documents"""
+        api_url = "documents/"
+        resp = self._make_request(api_url)
+        return ResultsList(result_type="document", **resp.json())
+
+    def document(self, document_id: int):
+        """details for a single document"""
+        api_url = f"documents/{document_id}/"
+        resp = self._make_request(api_url)
+        return to_namedtuple("document", resp.json())
+
+    def document_export(
+        self, document_id: int, transcription_id: int, include_images: bool = False
+    ):
+        """request a document export be compiled for download"""
+        api_url = f"documents/{document_id}/export/"
+        # export form requires a region_types list, which
+        # is a based on block type ids for this document
+        # and may also include "Undefined" and "Orphan"
+        document_info = self.document(document_id)
+        block_types = [block.pk for block in document_info.valid_block_types]
+        block_types.extend(["Undefined", "Orphan"])
+
+        data = {
+            "transcription": transcription_id,  # could be multiple(?)
+            "file_format": "alto",
+            "include_images": True,
+            "region_types": block_types,
+        }
+
+        resp = self._make_request(api_url, method="POST", data=data)
+        return to_namedtuple("status", resp.json())
+
+    def export_file_url(
+        self,
+        user_id: int,
+        document_id: int,
+        document_name: str,
+        file_format: str,
+        creation_time: datetime.datetime,
+    ) -> str:
+        """Generate URL for an export file generated by :meth:`document_export`"""
+        # is there any way get current user id from api based purely on token?
+        # if not, consider using users api endpoint to get numeric id from username
+
+        # ... get document name from api by document id?
+
+        # NOTE: filename logic copied & adapted from escriptorium
+        # import.export.BaseExporter logic
+        # NOTE2: escriptorium code uses datetime.now() so there's no guarantee
+        # this will match the completion time of the task...
+        base_filename = "export_doc%d_%s_%s_%s" % (
+            document_id,
+            slugify(document_name).replace("-", "_")[:32],
```

Comment:
Can you explain this in more detail? Using `slugify` seems like overkill, or at the very least confirming which characters are being removed and/or replaced.